Terminal.Gui/Core/TextFormatter.cs

```diff
@@ -0,0 +1,578 @@
+							break;
+						}
+					}
+					i++;
+				}
+			}
+
+			if (hot_key != (Rune)0 && hot_pos != -1) {
+				hotPos = hot_pos;
+
+				if (hot_key.IsValid && char.IsLetterOrDigit ((char)hot_key)) {
+					hotKey = (Key)char.ToUpperInvariant ((char)hot_key);
+					return true;
+				}
+			}
+
+			hotPos = -1;
+			hotKey = Key.Unknown;
+			return false;
+		}
+
+		/// <summary>
+		/// Replaces the Rune at the index specfiied by the <c>hotPos</c> parameter with a tag identifying 
+		/// it as the hotkey. 
+		/// </summary>
+		/// <param name="text">The text to tag the hotkey in.</param>
+		/// <param name="hotPos">The Rune index of the hotkey in <c>text</c>.</param>
+		/// <returns>The text with the hotkey tagged.</returns>
+		/// <remarks>
+		/// The returned string will not render correctly without first un-doing the tag. To undo the tag, search for 
+		/// Runes with a bitmask of <c>otKeyTagMask</c> and remove that bitmask.
+		/// </remarks>
+		public ustring ReplaceHotKeyWithTag (ustring text, int hotPos)
+		{
+			// Set the high bit
+			var runes = text.ToRuneList ();
+			if (Rune.IsLetterOrNumber (runes [hotPos])) {
+				runes [hotPos] = new Rune ((uint)runes [hotPos] | HotKeyTagMask);
+			}
+			return ustring.Make (runes);
+		}
+
+		/// <summary>
+		/// Removes the hotkey specifier from text.
+		/// </summary>
+		/// <param name="text">The text to manipulate.</param>
+		/// <param name="hotKeySpecifier">The hot-key specifier (e.g. '_') to look for.</param>
+		/// <param name="hotPos">Returns the postion of the hot-key in the text. -1 if not found.</param>
+		/// <returns>The input text with the hotkey specifier ('_') removed.</returns>
+		public static ustring RemoveHotKeySpecifier (ustring text, int hotPos, Rune hotKeySpecifier)
+		{
+			if (ustring.IsNullOrEmpty (text)) {
+				return text;
+			}
+
+			// Scan 
+			ustring start = ustring.Empty;
+			int i = 0;
+			foreach (Rune c in text) {
+				if (c == hotKeySpecifier && i == hotPos) {
+					i++;
+					continue;
+				}
+				start += ustring.Make (c);
+				i++;
+			}
+			return start;
+		}
+
+		/// <summary>
+		/// Draws the text held by <see cref="TextFormatter"/> to <see cref="Application.Driver"/> using the colors specified.
+		/// </summary>
+		/// <param name="bounds">Specifies the screen-relative location and maximum size for drawing the text.</param>
+		/// <param name="normalColor">The color to use for all text except the hotkey</param>
+		/// <param name="hotColor">The color to use to draw the hotkey</param>
+		public void Draw (Rect bounds, Attribute normalColor, Attribute hotColor)
+		{
+			// With this check, we protect against subclasses with overrides of Text
+			if (ustring.IsNullOrEmpty (text)) {
+				return;
+			}
+
+			Application.Driver.SetAttribute (normalColor);
+
+			// Use "Lines" to ensure a Format (don't use "lines"))
+			for (int line = 0; line < Lines.Count; line++) {
+				if (line > bounds.Height)
+					continue;
+				var runes = lines [line].ToRunes ();
```

Comment:
lines don't have the actual text value and that why the Label isn't changed correctly in the TimeAndDate.cs.

Reply:
My apologies for missing this comment. I did not see it. I think I now understand and will work on a fix.